File: src/main/java/com/yerti/core/prototype/command/BukkitCommandExtension.java
package com.yerti.core.prototype.command;

import org.bukkit.command.CommandException;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;

import java.util.Arrays;

public class BukkitCommandExtension extends org.bukkit.command.Command {

    private BukkitCommand command;
    private CommandExecutor executor;

    protected BukkitCommandExtension(BukkitCommand command, CommandExecutor executor) {
        super(command.getName(), command.getDescription(), command.getUsage(), Arrays.asList(command.getAliases()));

        this.command = command;
        this.executor = executor;
    }

    @Override
    public boolean execute(CommandSender commandSender, String s, String[] strings) {

        try {
            executor.onCommand(commandSender, this, s, strings);
        } catch (CommandException e) {
            e.printStackTrace();
        }

        return false;
    }
}

File: src/main/java/com/yerti/core/command/BukkitCommandExtension.kt
package com.yerti.core.command

import org.bukkit.command.Command
import org.bukkit.command.CommandException
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import java.util.*

class BukkitCommandExtension(private val command: BukkitCommand, private val executor: CommandExecutor) : Command(command.name, command.description, command.usage, Arrays.asList(*command.aliases)) {
    override fun execute(commandSender: CommandSender, s: String, strings: Array<String>): Boolean {
        try {
            executor.onCommand(commandSender, this, s, strings)
        } catch (e: CommandException) {
            e.printStackTrace()
        }
        return false
    }

}